### PR TITLE
feat: schema change of service program

### PR DESCRIPTION
**Pull request by jaspret1**
## Change Description
 I have changed the schema of the service program with one table and I changed get, add, put, delete, getById, getList, pagination, and searching API 
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist: (Check only applicable)

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/pages/fleet/maintenance/service-program/service-program-list/service-program-list.component.ts [hunk 1]
```diff
@@ -116,19 +114,19 @@ export class ServiceProgramListComponent implements OnInit {
...
-        complete: () => { },
-        error: () => { },
-        next: (result: any) => {
-          this.suggestions = result;
-        }
-      })
-    }
-  }, 800)
+  // getSuggestions = _.debounce(function (searchvalue) {
+  //   this.suggestions = [];
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/maintenance/service-program/service-program-list/service-program-list.component.ts]: why this function is commented?

File: src/app/pages/fleet/maintenance/service-program/add-service-program/add-service-program.component.ts [hunk 2]
```diff
@@ -19,10 +19,10 @@ export class AddServiceProgramComponent implements OnInit, AfterViewInit {
   vehicleModal: boolean = false;
   vehicles: any;
   tasks: any;
-  programID = '';
+  programID = [];
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/maintenance/service-program/add-service-program/add-service-program.component.ts:22]: this is not required

**Review by FH-sandeepkaur1 — CHANGES_REQUESTED** [review]

File: src/app/pages/fleet/maintenance/service-program/service-program-list/service-program-list.component.ts (hunk 1 above)
Comment (jaspret1) [reply src/app/pages/fleet/maintenance/service-program/service-program-list/service-program-list.component.ts]: now I have uncommented this code

File: src/app/pages/fleet/maintenance/service-program/add-service-program/add-service-program.component.ts (hunk 2 above)
Comment (jaspret1) [reply src/app/pages/fleet/maintenance/service-program/add-service-program/add-service-program.component.ts:22]: i have removed that code

**Review by FH-sandeepkaur1 — APPROVED** [review]

**Pull request merged**